diesel/src/query_builder/collected_query.rs

```diff
@@ -0,0 +1,48 @@
+use super::{AstPass, MoveableBindCollector, Query, QueryFragment, QueryId};
+use crate::backend::{Backend, DieselReserveSpecialization};
+use crate::result::QueryResult;
+use crate::sql_types::Untyped;
+
+#[derive(Debug)]
+#[must_use = "Queries are only executed when calling `load`, `get_result` or similar."]
+/// A SQL query variant with already collected bind data which can be moved
+pub struct CollectedQuery<T> {
```

Comment:
I would like to gate that behind the `i-implement-a-third-party-backend-and-opt-into-breaking-changes` feature, as it's not useful for anyone outside of diesel-async/other crates doing really low level stuff.